### ISSUE TITLE
jsx-indent on lines with JSX expressions

### ISSUE THREAD

**Issue by NiGhTTraX** [yannickcr/eslint-plugin-react#1176]
```jsx
const Item = ({ id, name, onSelect }) => <div onClick={onSelect}>
  {id}: {name}
</div>;
```

reports

```
Expected indentation of 2 space characters but found 0
```

It also complains about multiple variations:

JSX | Error
------|-------
`{name} foobar` | Expected indentation of 2 space characters but found 1 (react/jsx-indent)
`{id}: {name}` | Expected indentation of 2 space characters but found 0 (react/jsx-indent)
`{id} - {name}` | Expected indentation of 2 space characters but found 1 (react/jsx-indent)
`{id}. {name}` | Expected indentation of 2 space characters but found 0 (react/jsx-indent)
`{id} {name}` | No error
`{id}[2 spaces here]{name}` | Multiple spaces found before '{'  (no-multi-spaces) <- unrelated
`{id}[line break here]{name}` | No error
`{id}{name}` | No error

**Comment by ljharb** [comment]
What about if you have explicit whitespace instead?:
```jsx
const Item = ({ id, name, onSelect }) => (
  <div onClick={onSelect}>
    {id}
    {': '}
    {name}
  </div>
);
```

**Comment by NiGhTTraX** [comment]
@ljharb yes, that works, but it's not something I'd like to do.

**Comment by ljharb** [comment]
Thanks for clarifying.

While it's decidely a best practice to make whitespace explicit instead of relying on jsx to handle it intuitively, your original case is clearly a bug.

**Comment by golopot** [comment]
I cannot reproduce this. Seemed to be fixed.

**Comment by ljharb** [comment]
If so, we can close this with a PR containing the test case.

**Comment by ljharb** [comment]
@golopot the test case seems to fail, so this still seems like an issue.
